extensions/interactions/rule-input-defs.ts

```diff
@@ -196,6 +196,10 @@ export interface PencilCodeEditorRuleInputs {
   x: string;
 }
 
+export interface RatioInputRulesInputs {
+  x: string;
```

Comment:
Ditto here--should be a structure, not a string.

Reply:
done